### PR TITLE
feat: setup map context

### PR DESCRIPTION
**Pull request by akhelouat**
- Setup vcmap-core
- Adding 2d 3d map context. [GSREN3D-149](https://jira.camptocamp.com/browse/GSREN3D-149)

image

### CODE REVIEW

File: src/components/map/MapContext.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    class="map_container absolute overflow-hidden flex-1 inset-0 bg-white"
+  ></div>
+</template>
+
+<style scoped>
+.map_container {
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 8 8'%3E%3Cg fill='%23acacac' fill-opacity='0.4'%3E%3Cpath fill-rule='evenodd' d='M0 0h4v4H0V0zm4 4h4v4H4V4z'/%3E%3C/g%3E%3C/svg%3E");
+}
+
+:deep(.mapElement) {
```
Comment (fgravin) [inline src/components/map/MapContext.vue:23]: I will rework the global layout of the page, but this is not a good practice.
the map should take the full width/height and its container should set up the size

File: src/services/vcmap.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+const clearTargetElement = (target: string) => {
+  const targetElem: HTMLElement | null = document.getElementById(target)
+  if (targetElem) {
+    while (targetElem.firstChild) {
+      targetElem.removeChild(targetElem.firstChild)
+    }
+  }
+}
+
+export default async function initMap(
```
Comment (fgravin) [inline src/services/vcmap.ts:81]: I dont understand the divergences with https://github.com/virtualcitySYSTEMS/map-core-demo/blob/dbde70bdefa5e6ae252a8545ec9a7068e5ef8d50/src/api/api.js#L27

Could explain why you haven't just copied this file ?

File: src/services/vcmap.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      await collection.setActiveMap([...collection][0].name)
+      await collection.activeMap.gotoViewpoint(context.startingVP)
+    })
+  )
+
+  context.gotoViewpoint = async (vp) => {
+    await Promise.all([context.activeMap.gotoViewpoint(vp)])
+  }
+
+  window.mapContext = context
```
Comment (fgravin) [inline src/services/vcmap.ts:115]: really not the way the share objects.
also, if you create 2 maps, the first one reference will be lost.

**Review by fgravin — CHANGES_REQUESTED** [review]
Thanks for the PR.
Some minor comments and an important question about map-core-demo.

Also, the design of your component and your service is not well structured.
The map is not shared the proper way.

**Comment by fgravin** [comment]
@akhelouat you pushed the data, never do that on github it's not meant for that.
Also I can't review your second commit cause it's too large, please remove the `/data` folder and push again, Thanks

**Comment by akhelouat** [comment]
@fgravin it's done

File: map.config.json [hunk 4]
```diff
@@ -0,0 +1,1073 @@
+{
```
Comment (fgravin) [inline map.config.json:1]: good idea to have a config, but it's way too complex though, you should have the minimal config to make it work and understand how the map is built.

File: src/components/map/MapContext.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import initMap from '../../services/vcmap'
+
+onMounted(() => {
+  initMap(false, { x: -1.67, y: 48.1147, z: 16000, pitch: -10 }, 'map')
+})
+
+const is3d = ref(true)
+
+const toggle3d = () => {
+  initMap(is3d.value, { x: -1.67, y: 48.1147, z: 16000, pitch: -10 }, 'map')
```
Comment (fgravin) [inline src/components/map/MapContext.vue:12]: you should not init the map again and again while toggling 3D mode, the `mapCollection` is used for that with the `setActiveMap` method

File: src/services/vcmap.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    new TerrainLayer({
+      name: 'terrain',
+      url: 'data/terrain',
+    })
+  )
+
+  layers.add(
+    new CesiumTilesetLayer({
+      name: 'building',
+      url: 'data/buildings',
```
Comment (fgravin) [inline src/services/vcmap.ts:45]: should be http urls for now

**Review by fgravin — COMMENTED** [review]
Thanks for the update.
Unfortunately, the code is too approximative

- lot of useless code
- lack of architecture
- lack of understanding

It was easier to propose something almost from scratch in https://github.com/camptocamp/rennes3d-front/pull/11
I used some of your code though, thanks

I am closing this one.

**Comment by akhelouat** [comment]
@fgravin Thanks for the review I will try to improve

**Pull request closed without merging**